### PR TITLE
[Minecraft] Add new and improved CurseForge generic egg

### PR DESCRIPTION
**Pull request by dhkatz**
# Description

I have improved the CurseForge egg based on the work I did for the [Modrinth PR](https://github.com/parkervcp/eggs/pull/2229)

The old CurseForge egg only supported Forge. This new one supports Forge, Fabric, and Quilt. It also handles unpacking server packs better (ensuring files get moved into root of server)

[Here is a gist so you can see the installation script by itself](https://gist.github.com/dhkatz/edfa7f2efd6f819d0884760602daadd8#file-curseforge-sh)

Here are the startup args:

```
java $([[ -f user_jvm_args.txt ]] && printf %s "@user_jvm_args.txt") -Xms128M -Xmx{{SERVER_MEMORY}}M -Dterminal.jline=false -Dterminal.ansi=true $([[ ! -f unix_args.txt ]] && printf %s "-jar `cat .serverjar`" || printf %s "@unix_args.txt")
```

I did remove the current CurseForge egg because it doesn't make sense to support two eggs which do the same thing (and I placed this one in its own folder in `/java/curseforge` instead of in `/java/forge/curseforge-generic`)

## Checklist for all submissions

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

## New egg Submissions

1. [x] Does your submission pass tests (server is connectable)?
3. [x] Have you added the egg to the main README.md and any other README files in subdirectories of the egg (e.g /game_eggs) according to the alphabetical order?
4. [x] Have you added a unique README.md for the egg you are adding according to the alphabetical order?
5. [x] You verify that the start command applied does not use a shell script
6. [x] The egg was exported from the panel

### CODE REVIEW

**Comment by parkervcp** [comment]
The generic moniker was because it wasn't a single pack that was supported.

**Comment by dhkatz** [comment]
To me a CurseForge egg that only supports one pack doesn't make sense anyways

**Comment by dhkatz** [comment]
If you want me to change anything about naming just let me know 😊

**Comment by parkervcp** [comment]
> If you want me to change anything about naming just let me know blush

I am going to review the egg today when I have a chance.

**Comment by dhkatz** [comment]
Just tested with the issue reported in https://github.com/parkervcp/eggs/issues/2157

It seems that the current way of checking the file id did not work for very old versions of packs because they don't appear in the `.data.latestFiles[]` so I am now instead fetching the file data directly

This script also does automatically check if the unzipped pack places all files in the root folder of if it unpacks into a second directory. So it solves the issue reported there

In other words both these pack structures are supported:

-  `/mnt/server/`
    -  `/mods`
    - `/config`
    - etc.
- `/mnt/server/somecoolpack` (File structure is transformed to the first one) ^
    - `/mods`
    - `/config`
    - etc.

**Comment by parkervcp** [comment]
Funny enough `${SERVER_DIR}` uses more characters than `/mnt/server` so replacing it with a variable really doesn't save any.

still reviewing the rest.

**Comment by parkervcp** [comment]
@dhkatz have you tried ATM7 or ATM8 with this?

**Comment by dhkatz** [comment]
> Funny enough `${SERVER_DIR}` uses more characters than `/mnt/server` so replacing it with a variable really doesn't save any.
> 
> still reviewing the rest.

I used that for easy testing, the script runs outside of Pterodactyl too

> @dhkatz have you tried ATM7 or ATM8 with this?

Yes, I actually am running ATM7 server in my own Pterodactyl instance using this that I play on with friends. I did also test ATM8 and it seemed to work fine too (although ATM8 really eats RAM yikes).

**Comment by parkervcp** [comment]
So I have checked it out and it looks good.

**Review by parkervcp — APPROVED** [review]
LGTM

**Comment by parkervcp** [comment]
Thank you for the change
